chia/wallet/util/wallet_sync_utils.py

```diff
@@ -327,13 +361,12 @@ async def fetch_header_blocks_in_range(
         if res_h_blocks_task is not None:
             log.debug(f"Using cache for: {start}-{end}")
             if res_h_blocks_task.done():
-                res_h_blocks: Optional[RespondHeaderBlocks] = res_h_blocks_task.result()
+                res_h_blocks: Optional[Union[RespondBlockHeaders, RespondHeaderBlocks]] = res_h_blocks_task.result()
             else:
                 res_h_blocks = await res_h_blocks_task
         else:
             log.debug(f"Fetching: {start}-{end}")
-            request_header_blocks = RequestHeaderBlocks(request_start, request_end)
-            res_h_blocks_task = asyncio.create_task(_fetch_header_blocks_inner(all_peers, request_header_blocks))
+            res_h_blocks_task = asyncio.create_task(_fetch_header_blocks_inner(all_peers, request_start, request_end))
             peer_request_cache.add_to_block_requests(request_start, request_end, res_h_blocks_task)
             res_h_blocks = await res_h_blocks_task
```

Comment:
it looks odd to `await` the request task inside the loop. Why do we create a task for each request unless we want to request all block ranges in parallel? (which I would expect us to want)

Reply:
We create a task because we might have two async io threads that are requesting the same info at the same time, and we want to cache the task so they can both obtain the result when it arrives.